gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/selector/OfflineBaseMapSelectorViewModel.java

```diff
@@ -38,39 +36,39 @@
     FAILURE
   }
 
-  @Hot private final FlowableProcessor<OfflineBaseMap> downloadClicks = PublishProcessor.create();
-  private final LiveData<Event<DownloadMessage>> messages;
+  @Hot private final FlowableProcessor<OfflineBaseMap> baseMapDownloads = PublishProcessor.create();
+
+  @Hot(terminates = true, errors = false)
+  private final Single<DownloadMessage> downloadMessage;
   private final OfflineUuidGenerator offlineUuidGenerator;
 
   @Inject
   OfflineBaseMapSelectorViewModel(
       OfflineBaseMapRepository offlineBaseMapRepository,
       OfflineUuidGenerator offlineUuidGenerator) {
-    this.messages =
-        LiveDataReactiveStreams.fromPublisher(
-            downloadClicks.switchMapSingle(
-                baseMap ->
-                    offlineBaseMapRepository
-                        .addAreaAndEnqueue(baseMap)
-                        .toSingleDefault(DownloadMessage.STARTED)
-                        .onErrorReturn(this::onEnqueueError)
-                        .map(Event::create)));
+    this.downloadMessage =
+        baseMapDownloads
```

Comment:
I find the interaction between values and completion in `baseMapDownloads`, the `Completable` returned by `addAreaAndEnqueue`, and the default value set by `toSingleDefault` hard to grok. Would things be simpler if we didn't rely on termination at all, and just used an `Observable` for `downloadMessage`? We can and should just debounce (or rather, only take the first) download button click.

Reply:
I do think it would make it simpler in a certain sense, but I also feel using an `Observable` for this purpose and only taking the first emission is sort of like a square peg round hole type of situation.

From an outside reader's perspective w/o context, I actually feel using an `Observable` that we only pull a single emission from would give me more pause, I'd likely think "why not just use a single?" and then we'd have special explanation around why we're using less specific types, etc.

What I do think makes sense here is using a SingleSubject for the `baseMapDownloads` instead of a Publisher, as then we wouldn't need to call `complete` explicitly. Maybe we can see if that makes it clearer? We'd effectively get the debouncing behavior for free since late observers just receive the initial terminal event. (the behavioral effect of this iiuc, would be that if a user clicks download, has insane finger speed and beats the UI and quickly goes to another area and clicks it again, only the first area will be downloaded).

Lemme take another stab at it using these types and see if I can simplify, if not we can swap em out for an observer.

Reply:
Ok, I tried this out, but admittedly it's not that different, we only remove the explicit call to `complete` and the SingleSubject better communicates that we only care about the first basemap selected, so lmk if you feel strongly about using an `Observable` instead here and we can make the switch.

Reply:
I think in this case the question of whether to use `Single` or `Observable` comes down to levels of abstraction, or more specifically, whether the VM is aware of / models the fact that it's a single-use object. Also, experience with Rx in Android so far has shown that at the level of lifecycle-aware objects, it's almost always better to make no or few assumptions about the order in which things happen. It's also slightly easier to reason about "the VM will emit download status messages for the View to display" than "the VM will emit a single message when the download is complete". How/when the messages are generated could be considered a detail of the current flow; adding the ability a message when the download starts wouldn't break the former contract, but it would break the latter.

tl;dr: I'm wary of storing references to streams that terminate inside life-cycle aware objects due additional complexity due to lifecycle x termination state. I can't think of a good example now, so if you feel strongly please feel free to keep it for now till I can think of one!